torch/distributed/elastic/rendezvous/c10d_rendezvous_backend.py

```diff
@@ -0,0 +1,195 @@
+# Copyright (c) Facebook, Inc. and its affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+import binascii
+import codecs
+import logging
+import os
+from datetime import timedelta
+from typing import Any, Optional, Tuple, cast
+
+from torch.distributed import Store, TCPStore
+
+from .api import RendezvousConnectionError, RendezvousParameters, RendezvousStateError
+from .default_rendezvous import RendezvousBackend, Token
+from .utils import _matches_machine_hostname, _parse_rendezvous_endpoint
+
+log = logging.getLogger(__name__)
+
+
+class C10dRendezvousBackend(RendezvousBackend):
+    """Represents a C10d-backed rendezvous backend.
+
+    Args:
+        store:
+            The `Store` instance to use to communicate with the C10d store.
+        run_id:
+            The run id of the rendezvous.
+    """
+
+    # See the explanation in the __init__ method.
+    _NULL_SENTINEL = "Y2FuaW1hZGFt"
```

Comment:
Can we use something like "<null>" for this instead or is it likely the rendezvous state might consist of something like this?

Reply:
I wanted to have something that is practically unique. Actually `Y2FuaW1hZGFt` means "my dear Ada" (my daughter's name) in Turkish in base64-encoded form :) So very unlikely that it will match a binary string produced by Python's pickle. I thought about using an empty string, but I wasn't sure whether it would be always an invalid state value (what if we optimize our state serialization logic in the future and use "" for initial state). Instead of taking the risk, I just came up with this "random" string.